src/manage_nsfs/nc_lifecycle.js

```diff
@@ -498,6 +509,63 @@ class NCLifecycle {
     /////////////////////////////////////////////
     //////// NON CURRENT VERSION HELPERS ////////
     /////////////////////////////////////////////
+    /**
+     * load versions list bacth and update state for the next cycle
+     * @param {Object} object_sdk
+     * @param {Object} bucket_json
+     * @param {Object} rule_state
+     * @returns
+     */
+    async load_versions_list(object_sdk, bucket_json, rule_state) {
+        const list_versions = await object_sdk.list_object_versions({
+            bucket: bucket_json.name,
+            prefix: rule_state.filter?.prefix,
+            limit: config.NC_LIFECYCLE_LIST_BATCH_SIZE,
+            key_marker: rule_state.key_marker_versioned,
+            version_id_marker: rule_state.version_id_marker
+        });
+        if (list_versions.is_truncated) {
+            rule_state.is_finished = false;
+            rule_state.key_marker_versioned = list_versions.next_marker;
+            rule_state.version_id_marker = list_versions.next_version_id_marker;
+        } else {
+            rule_state.is_finished = true;
+        }
+        const bucket_state = this.lifecycle_run_status.buckets_statuses[bucket_json.name].state;
```

Comment:
not necessarily on this PR if you don't want, I think we can move the accumulation of num_processed_objects to update_status(), can you add it as a gap?